### DISCUSSION TITLE
Baking a tool into a Renovate container image

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#22240] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab EE

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

An internal tool is required when running a postUpdate task on a self hosted Renovate runner.
The tool is built on a non Ubuntu operating system. 

How would you suggest to go about it?

### What you tried so far.

As of now, no idea how to solve it.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
The Renovate images are built on top of Ubuntu 20.04, so you need the tool to support that.

Something like:

```
FROM renovate/renovate
USER 0
# install things here
USER 1000
```

**Reply by lmilbaum** [reply]
Thanks @rarkins. Replaced Alpine with Ubuntu. The needed package is internal one which is built and packaged for RHEL.

**Reply by rarkins** [reply, marked as answer]
In that case it sounds like you need to create your own image:
- Based on RHEL
- Install the languages and package managers you need (node, npm, go, etc)
- Install your tool
- Install Renovate using npm

**Reply by lmilbaum** [reply]
Yup. That is a good approach. Thank you.

**Reply by lmilbaum** [reply]
Food for thought. Provide `containerbase/base` builds for multiple operating systems. Tags will be used for picking non-default one.

**Reply by rarkins** [reply]
Unfortunately far too much work to do so

**Comment by lmilbaum** [comment]
cc @yerdreic
